src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/iOS/WebUI.cs

```diff
@@ -44,6 +48,25 @@ public WebUI(IPlatformParameters parameters)
             {
                 throw new ArgumentException("parameters should be of type PlatformParameters", "parameters");
             }
+
+            this.didEnterBackgroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidEnterBackgroundNotification, OnMoveToBackground);
+            this.willEnterForegroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.WillEnterForegroundNotification, OnMoveToForeground);
+        }
+
+        void OnMoveToBackground(NSNotification notification)
+        {
+            //After iOS 11.3, it is neccesary to keep a background task running while moving an app to the background in order to prevent the system from reclaiming network resources from the app. 
+            //This will prevent authentication from failing while the application is moved to the background while waiting for MFA to finish.
+            this.taskId = UIApplication.SharedApplication.BeginBackgroundTask(() => {
+                if (this.taskId != UIApplication.BackgroundTaskInvalid)
+                    UIApplication.SharedApplication.EndBackgroundTask(this.taskId);
```

Comment:
this should set taskId to BackgroundTaskInvalid too

Reply:
fixed